plugins/user-authenticators/ldap/src/main/java/org/apache/cloudstack/ldap/LdapManagerImpl.java

```diff
@@ -257,6 +258,19 @@ public LdapUserResponse createLdapUserResponse(final LdapUser user) {
 
     @Override
     public LdapConfigurationResponse deleteConfiguration(final LdapDeleteConfigurationCmd cmd) throws InvalidParameterValueException {
+        Long id = cmd.getId();
+        String hostname = cmd.getHostname();
+        if (id == null && StringUtils.isEmpty(hostname)) {
+            throw new InvalidParameterValueException("Either id or hostname must be specified");
+        }
+        if (id != null) {
```

Comment:
what if id and hostname/domainid/port are both passed ?

Reply:
it'll only consider id for searching the ldap config.